core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionDetails.java

```diff
@@ -125,18 +125,6 @@ default String getRootPath() {
     return null;
   }
 
-  /**
-   * Sets the root folder path, given as a string.
-   * <p>
-   * An empty root folder path value should be converted to {@code null}.
-   * Further syntax validation is performed only after variable substitution.
-   * <p>
-   * The default interface implementation exists to ensure backward compatibility and does nothing.
-   *
-   * @param rootPath The root path.
-   */
-  default void setRootPath( String rootPath ) { }
```

Comment:
This is not needed in the interface, after all, unlike what was first thought. Only the implementing classes need to have the setter. Used via reflection by the meta-store and rest/jackson deserialization.